desktop/cmp/panel/impl/PanelHeader.ts

```diff
@@ -41,57 +62,57 @@ export const panelHeader = hoistCmp.factory({
+        // Return null if nothing to display.
+        if (isNil(displayedTitle) && isNil(displayedIcon) && isEmpty(displayedHeaderItems)) {
+            return null;
+        }
+
+        // Return a vertically oriented header if collapsed to left or right side.
+        if (isCollapsedToLeftOrRight) {
+            return vbox({
+                className: classNames(className, sideCls, compactCls),
+                flex: 1,
                 items: [
-                    icon || null,
+                    displayedIcon,
                     displayedTitle
                         ? box({
                               className: titleCls,
                               flex: 1,
                               item: span({className: `${titleCls}__inner`, item: displayedTitle})
                           })
                         : filler(),
-                    hbox({
-                        className: 'xh-panel-header__items',
-                        items: [
-                            ...(!collapsed || isModal ? headerItems : []),
-                            modalButton({panelModel}),
-                            collapseButton({panelModel})
-                        ],
-                        onDoubleClick: e => e.stopPropagation()
-                    })
+                    ...displayedHeaderItems
                 ],
                 onDoubleClick
             });
         }
 
-        // 2) ...otherwise its a narrow, sidebar
-        return vbox({
-            className: classNames(className, sideCls, compactCls),
-            flex: 1,
+        // Return a standard horizontal header otherwise.
+        // Panel is expanded, modal, and/or collapsed to the top or bottom side.
+        return hbox({
+            className: classNames(className, compactCls),
             items: [
-                collapseButton({panelModel}),
-                icon || null,
+                displayedIcon,
                 displayedTitle
                     ? box({
                           className: titleCls,
+                          flex: 1,
                           item: span({className: `${titleCls}__inner`, item: displayedTitle})
                       })
-                    : null
+                    : filler(),
+                hbox({
+                    className: 'xh-panel-header__items',
+                    items: displayedHeaderItems,
+                    onDoubleClick: e => e.stopPropagation()
+                })
             ],
             onDoubleClick
         });
     }
 });
 
-const collapseButton = hoistCmp.factory(({panelModel}) => {
-    const {showHeaderCollapseButton, collapsible, isModal} = panelModel as PanelModel;
-    if (!showHeaderCollapseButton || !collapsible || isModal) return null;
-
-    const {vertical, collapsed, contentFirst} = panelModel,
+const collapseToggleButton = hoistCmp.factory(({panelModel}) => {
+    const {vertical, collapsed, contentFirst} = panelModel as PanelModel,
```

Comment:
Great that we could move this visibility logic out of these internal components (and completely remove the wrapper below) :)